phpBB/phpbb/storage/adapter/local.php

```diff
@@ -221,12 +242,68 @@ protected function ensure_directory_exists($path)
 		}
 	}
 
+	/**
+	 * Removes the directory tree ascending until it finds a non empty directory.
+	 *
+	 * @param string	$path	The file path
+	 */
+	protected function remove_empty_dirs($path)
+	{
+		$dirpath = dirname($this->root_path . $path);
+		$filepath = dirname($this->root_path . $this->get_path($path) . $this->get_filename($path));
+		$path = filesystem_helper::make_path_relative($filepath, $dirpath);
+
+		do
+		{
+			$parts = explode('/', $path);
+			$parts = array_slice($parts, 0, -1);
+			$path = implode('/', $parts);
+		}
+		while ($path && @rmdir($dirpath . '/' . $path));
+	}
+
+	/**
+	 * Get the path to the file, appending subdirectories for directory depth
+	 * if $dir_depth > 0.
+	 *
+	 * @param string	$path	The file path
+	 */
+	protected function get_path($path)
+	{
+		$dirname = dirname($path);
+
+		$hash = md5(basename($path));
+
+		$parts = str_split($hash, 2);
+		$parts = array_slice($parts, 0, $this->dir_depth);
+
+		// Create path
+		$path = $dirname . DIRECTORY_SEPARATOR;
+
+		if (!empty($parts))
```

Comment:
if (isset($parts[0]))

Reply:
is not the same?

Reply:
Yes, isset is only a little bit faster. And through the md5 ``$parts[0]`` is never empty if exists.

Reply:
I have test and is almost the same, in fact, empty is a little faster